### PR TITLE
Char Module and tests

### PR DESCRIPTION
**Pull request by rascala**
Hi Karl,

I added a basic System.Char module and some associated tests.
Please have a look.

### CODE REVIEW

**Comment by rascala** [comment]
Woops, actually, String.collect wont work.
The test had to be commented out for the string() issue and I missed it.

I will fix it soon

**Comment by rascala** [comment]
Would a set be faster?

**Comment by kjnilsson** [comment]
not faster than a map on recent versions of erlang no.

**Comment by rascala** [comment]
Done, can you have a look?

File: src/Microsoft.FSharp.Core.Operators.erl [hunk 1]
```diff
@@ -44,6 +44,17 @@ seq({seq, _} = X) -> X;
 seq(X) when is_list(X) ->
     {seq, fun () -> {list, X} end}.
 
+% string(X) when is_list(X) ->
```
Comment (kjnilsson) [inline src/Microsoft.FSharp.Core.Operators.erl]: remove commented code

File: src/System.Char.erl [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+% TODO, figure out what this means and implement for unicode
+'IsNumber'(Chr) ->
+	'IsDigit'(Chr).
+'IsNumber'(Str,Idx) ->
+	for_valid_index(fun 'IsNumber'/1, Str, Idx).
+
+%% phew! should generate the below -- unused, left here as documentation
+%% "!\"#،؍᠀᠁᠂᠃᠄᠅᠆᠇᠈᠉᠊〔〕〖〗〘〙〚〛〜〝〞〟%&'()*؛᥄᥅〰,-./؞؟᧞᧟〽:;٪٫٬٭᨞᨟゠?@۔᭚᭛᭜᭝᭞᭟᭠・[\\]܀܁܂܃܄܅܆܇܈܉܊܋܌܍‐‑‒–—―‖‗‘’‚‛“”„‟†‡•‣․‥…‧꡴꡵꡶꡷_߷߸߹‰‱′″‴‵‶‷‸‹›※‼‽‾‿⁀⁁⁂⁃﴾﴿{।॥⁅⁆⁇⁈⁉⁊⁋⁌⁍⁎⁏⁐⁑︐︑︒︓︔︕︖︗︘︙}॰⁓⁔⁕⁖⁗⁘⁙⁚⁛⁜⁝⁞︰︱︲︳︴︵︶︷︸︹︺︻︼︽︾︿﹀﹁﹂﹃﹄﹅﹆﹇﹈﹉﹊﹋﹌﹍﹎﹏﹐﹑﹒¡෴⁽⁾﹔﹕﹖﹗﹘﹙﹚﹛﹜﹝﹞﹟﹠﹡«๏๐๑๒๓๔๕๖๗๘๙๚๛₍₎﹣­༄༅༆༇༈༉༊་༌།༎༏༐༑༒〈〉﹨·༺༻༼༽❨❩❪❫❬❭❮❯❰❱❲❳❴❵﹪﹫»྅⟅⟆！＂＃¿࿐࿑⟦⟧⟨⟩⟪⟫％＆＇（）＊;၊။၌၍၎၏⦃⦄⦅⦆⦇⦈⦉⦊⦋⦌⦍⦎⦏⦐⦑⦒⦓⦔⦕⦖⦗⦘，－．／·჻⧘⧙⧚⧛：；՚՛՜՝՞՟፡።፣፤፥፦፧፨⧼⧽？＠։֊᙭᙮⳹⳺⳻⳼［＼］־᚛᚜⳾⳿＿׀᛫᛬᛭⸀⸁⸂⸃⸄⸅⸆⸇⸈⸉⸊⸋⸌⸍⸎⸏⸐⸑⸒⸓⸔⸕⸖⸗｛׃᜵᜶⸜⸝｝׆។៕៖、。〃｟｠｡｢｣､･׳״៘៙៚〈〉《》「」『』【】"
+% punctuationCharacterList() ->
+% 	lists:append([
```
Comment (kjnilsson) [inline src/System.Char.erl]: remove commented code

**Comment by kjnilsson** [comment]
Thanks!

**Pull request merged**